libs/crypto/src/mcl_dkg.cpp

```diff
@@ -396,6 +396,147 @@ std::vector<DkgKeyInformation> TrustedDealerGenerateKeys(uint32_t committee_size
   return output;
 }
 
+/**
+ * Generates a private key and the corresponding public key
+ *
+ * @param generator Choice of generator on the elliptic curve
+ * @return Pair of private and public keys
+ */
+std::pair<PrivateKey, PublicKey> GenerateKeyPair(Generator const &generator)
+{
+  std::pair<PrivateKey, PublicKey> key_pair;
+  key_pair.first.setRand();
+  key_pair.second.clear();
+  bn::G2::mul(key_pair.second, generator, key_pair.first);
+  return key_pair;
+}
+
+/**
+ * Computes a deterministic hash to the finite prime field from one public key and the set
+ * of all eligible notarisation keys
+ *
+ * @param notarisation_key Particular public key of a cabinet member
+ * @param cabinet_notarisation_keys Public keys of all cabinet members
+ * @return Element of prime field
+ */
+bn::Fr SignatureAggregationCoefficient(PublicKey const &             notarisation_key,
+                                       std::vector<PublicKey> const &cabinet_notarisation_keys)
+{
+  bn::Fr coefficient;
+  coefficient.clear();
+
+  // Reserve first 48 bytes for some fixed value for hygenic reuse of the
+  // hashing function
+  std::string concatenated_keys = "BLS Aggregation";
+  while (concatenated_keys.length() < 48)
+  {
+    concatenated_keys += "0";
+  }
```

Comment:
It would be worth to preallocate the `concatentated_keys` buffer to avoid expensive memory allocations

```suggestion
  std::string concatenated_keys = "BLS Aggregation";
  concatenated_keys.reserve(<insert estimation of final string>);
  while (concatenated_keys.length() < 48)
  {
    concatenated_keys.push_back('0');
  }
```

Reply:
Done!